internal/config/properties.go

```diff
@@ -2,55 +2,93 @@ package config
 
 import (
 	"context"
+	"fmt"
 	"github.com/keikoproj/iam-manager/pkg/k8s"
 	"github.com/keikoproj/iam-manager/pkg/log"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
 	"strings"
 )
 
-const (
-	iamPolicyWhitelist    = "iam.policy.action.prefix.whitelist"
-	iamPolicyBlacklist    = "iam.policy.resource.blacklist"
-	iamPolicyS3Restricted = "iam.policy.s3.restricted.resource"
-	awsAccountId          = "aws.accountId"
-	awsMasterRole         = "aws.MasterRole"
-	managedPolicies       = "iam.managed.policies"
-)
-
-const (
-	separator = ","
-)
+var Props *Properties
 
-//Properties struct loads the properties
 type Properties struct {
-	AllowedPolicyAction       []string
-	RestrictedPolicyResources []string
-	RestrictedS3Resources     []string
-	AWSAccountId              string
-	AWSMasterRole             string
-	ManagedPolicies           []string
-}
-
-//LoadProperties function loads properties from various sources
-func LoadProperties(ctx context.Context, kClient *k8s.Client, ns string, cmName string) *Properties {
-	log := log.Logger(ctx, "config", "LoadProperties")
-	log.WithValues("namespace", ns)
-	log.Info("loading properties")
-
-	res := kClient.GetConfigMap(ctx, ns, cmName)
-	allowedActions := strings.Split(res.Data[iamPolicyWhitelist], separator)
-	restrictedResources := strings.Split(res.Data[iamPolicyBlacklist], separator)
-	restrictedS3Resources := strings.Split(res.Data[iamPolicyS3Restricted], separator)
-	managedPolicies := strings.Split(res.Data[managedPolicies], separator)
-	awsAccountId := res.Data[awsAccountId]
-	awsMasterRole := res.Data[awsMasterRole]
-
-	return &Properties{
-		AllowedPolicyAction:       allowedActions,
-		RestrictedPolicyResources: restrictedResources,
-		RestrictedS3Resources:     restrictedS3Resources,
-		AWSAccountId:              awsAccountId,
-		AWSMasterRole:             awsMasterRole,
-		ManagedPolicies:           managedPolicies,
+	allowedPolicyAction             []string
+	restrictedPolicyResources       []string
+	restrictedS3Resources           []string
+	awsAccountID                    string
+	awsMasterRole                   string
+	managedPolicies                 []string
+	managedPermissionBoundaryPolicy string
+}
+
+func LoadProperties(cm *v1.ConfigMap) {
+	allowedPolicyAction := strings.Split(cm.Data[propertyIamPolicyWhitelist], separator)
+	restrictedPolicyResources := strings.Split(cm.Data[propertyIamPolicyBlacklist], separator)
+	restrictedS3Resources := strings.Split(cm.Data[propertyIamPolicyS3Restricted], separator)
+	managedPolicies := strings.Split(cm.Data[propertyManagedPolicies], separator)
+	awsAccountID := cm.Data[propertyAwsAccountID]
+	awsMasterRole := cm.Data[propertyAwsMasterRole]
+	managedPermissionBoundaryPolicy := fmt.Sprintf(IamManagedPermissionBoundaryPolicy, awsAccountID)
```

Comment:
So, we can make permission policy boundary name also as part of config map and users can provide the value instead of we hard coding it to "iam-manager-permission-boundary" and when you are setting the variable you can just do fit.sprintf and substitute with ARN if user provided just a name